### DISCUSSION TITLE
GraphQL get custom object field details

### DISCUSSION THREAD

**Discussion by etfz** [netbox-community/netbox#10277] [Q&A]
Hi,

I've created a custom field of type device. When retrieving custom field data using GraphQL, this field only returns the device ID. Is there any way I can retrieve the device *object* in the same query? Specifically I'm looking to retrieve the device's primary IP address.

**Comment by aopdal** [comment]
Hi,
I have just started to play with GraphQL, so I don't know if this does help you, or confuse you ;-)
To get the primary IP for a specific device this works for me: 
`query {
  device(id: 123) {
    name
    primary_ip4 {
      address
    }
  }
}
`
Result:
`{
  "data": {
    "device": {
      "name": "z9cx3",
      "primary_ip4": {
        "address": "172.18.24.3/24"
      }
    }
  }
}
`

Regards
Arne Opdal

**Reply by candlerb** [reply]
But that's not using a custom field.

**Reply by aopdal** [reply]
It was just an example how its possible to retrieve the device primary ip address when you have the device ID.

And it ended up as a source of confusion ;-) 

Perhaps someone can be more helpful if we get some more details in the usecase.

**Reply by etfz** [reply]
Thank you, but I'm afraid that is not what I'm looking for. The device ID is not known ahead of time but rather retrieved using a GraphQL query.
